### DISCUSSION TITLE
Windows Status

### DISCUSSION THREAD

**Discussion by RossSmyth** [serialport/serialport-rs#171] [General]
What is up with the open Windows PRs. Is there no one to review them? But there are some Windows-related PRs that were closed recently so someone reviewed those. Windows is second-class with this crate for sure right now. What ever happened from that meeting several months ago?

It is currently more sane for me to run Ubuntu in WSL 2, forward my FTDI serial port adapter via a TCP port to WSL 2 from Windows, expose the TCP port as a serial port to Linux, and talk to my device that way than to use this library on Windows itself.

I can probably review PRs if that is needed, though there are several people who've interacted here who could and probably can better than I. One person even emailed me directly because I made comments here expressing their frustration with the state of Windows in this library.

**Comment by jerrywrice** [comment]
Just dropped by the site for the first time in several months, and noticed your new discussion post. Let me chime in. I volunteered to assist with this crate in the latter half of 2023, offering assistance with testing, characterization, and fixes and/or development of the Windows version of the crate. I developed and shared a serial characterization/test application example, attended a couple of the occasional on-line project team meetings (which were sparsely attended), but found that there was almost no follow-up communications after the meetings. I eventually lost interest and moved on to other 'rust' things. I have the same perspective as you (RossSmyth) regarding the Windows platform apparently being the 'lonely step-child' platform of this crate. For me Windows has been a primary development and usage environment for 25+ years - with significant comm development in c/c++ and c# applications. I've also done a fair bit of Linux c++ comm development. As I'm sure you know, Windows has its quirks and warts, but nevertheless is a major platform in terms of overall user base size. I prefer using the serialport-rs serial crate, and based on other rust serial crates I've tried, it seems to have the most promise IMO. I'm (also) open to re-engaging and offering support to this project, but I need to have confidence the project's support team leadership will communicate and respond to activity related exchanges in a reasonable time-frame. Even a short response message stating that they've gotten diverted and must be away is better than no response at all. One other significant annoyance for me was the fact that the primary communications channel was Matrix. I understand its value for crate users to ask questions, whereby the eventual answers are publicly shared for all other users (including future ones) to read. For support members, I suggest once someone takes on a team sanctioned support task, that they and other directly involved team members should share email addresses. If someone wants to maintain their privacy they can provide a 'burner' email address for this purpose. Development/test/documentation related internal communications solely using the Matrix channel is highly ineffective - IMO.
